### ISSUE TITLE
Bug: Cant open serial ports anymore after pulling the newest code

### ISSUE THREAD

**Issue by fabianbernhart** [JadKHaddad/serial-vau#58]
### Debug Message
```
2024-11-09T20:54:44.708623Z DEBUG open_serial_port: serial_vau::app::serial_state: Opening serial port options=CoreOpenSerialPortOptions { initial_read_state: Read, baud_rate: 0, data_bits: Eight, flow_control: None, parity: None, stop_bits: One, timeout: 
0ns }
```

### Error Message

```
2024-11-09T20:54:44.728503Z ERROR open_serial_port: serial_vau::tauri_app::error: Failed to open serial port: Failed to open serial port: Failed to open port: Failed to open port: Ein an das System angeschlossenes Gerät funktioniert nicht.: Failed to open 
serial port: Failed to open port: Failed to open port: Ein an das System angeschlossenes Gerät funktioniert nicht.: Failed to open port: Failed to open port: Ein an das System angeschlossenes Gerät funktioniert nicht.: Failed to open port: Ein an das System angeschlossenes Gerät funktioniert nicht.: Ein an das System angeschlossenes Gerät funktioniert nicht.

```

#### _One_: `port.lastUsedOpenOptions` are wrong #15

```vue

/*src/components/serial-port/SerialPort.vue*/

<v-btn 
    @click="openSerialPort(
      port.name,
      port.lastUsedOpenOptions // this might be not set and then the defaults trigger
   )"
   variant="plain"
>
    Open
</v-btn>
```

I debuged the Ports and see they have `port.lastUsedOpenOptions` baudRate 0

![grafik](https://github.com/user-attachments/assets/96320b8e-d14b-4550-847c-8dcc8252eb7e)

**Comment by fabianbernhart** [comment]
@JadKHaddad whats the baudRate for this devices ...

**Comment by fabianbernhart** [comment]
this fixed dthe problem termpary please create an error/warning for devices opended with baudRate 0 cause the internet says that it is impossible that a bbaudRate is 0

![grafik](https://github.com/user-attachments/assets/aa245896-de54-499f-94e9-8e143485d5b5)

**Comment by JadKHaddad** [comment]
[Here](https://github.com/JadKHaddad/serial-vau/blob/main/src-tauri%2Fsrc%2Fapp%2Fserial_state%2Fmodel%2Fmod.rs#L57) defaulting to 0. Was fixed in #16 (115200)

**Comment by JadKHaddad** [comment]
It should Not be hardcoded. The User must be able to choose in the UI

**Comment by fabianbernhart** [comment]
> It should Not be hardcoded. The User must be able to choose in the UI

see the bold text **temporary** only so i can work :^)

**Comment by fabianbernhart** [comment]
Guess the bug occurs when you plug your device as a new initial state and there is no correct `lastUsedOpenOptions`

**Comment by fabianbernhart** [comment]
I see you also added an `tag `field to the `OpenSerialPotOptions`

**Comment by fabianbernhart** [comment]
The BaudRate for my Brain: 115200

**Comment by JadKHaddad** [comment]
Fixed in [21a4c8a](https://github.com/JadKHaddad/serial-vau/pull/56/commits/21a4c8ae0618108b1dd34496ec9263af2ea84f9f)

**Comment by JadKHaddad** [comment]
My other esp32-c3 works fine with baudrate 0. That's why I did not notice the bug. 

We don't need the temporary fix. But you were right.
